### DISCUSSION TITLE
Magneto 2 install > warden shell > composer update errors out

### DISCUSSION THREAD

**Discussion by kraiglarson** [wardenenv/warden#802] [Help & Support]
I've had this local dev instance of Magento up and running for a couple years. Just in the last couple days, I have been unable to run "composer update" or "composer install" from within warden shell. If I'm outside the container, on my Mac, I can run "composer install" just fine (without the following error). When inside the container, via "warden shell", I get errors like this:

```
www-data@sslocal-php-fpm:/var/www/html$ composer update --dry-run
Loading composer repositories with package information

  [Composer\Repository\InvalidRepositoryException]                                                                                           
  No valid composer.json was found in any branch or tag of https://github.com/fastly/fastly-magento2.git, could not load a package from it.  
```

It's as if, when I'm inside the container, composer can't reach out to grab anything on https://github.com/. If I remove "fastly/fastly-magento2" from my composer.json, then it just fails on the next module that uses a GitHub repo.

If I run this:
```
www-data@sslocal-php-fpm:/var/www/html$ ssh -T git@github.com
Hi kraiglarson! You've successfully authenticated, but GitHub does not provide shell access.
```
(I found this from a Google search, to check and see if a Docker container has SSH access to GitHub.)

I did recently update my MacBook Pro (M2 Pro chip) to Sonoma 14.7, and like Warden normally does, I had to re-authorize SSH. I'm using Docker Desktop v4.34.0. I also tried an update to Docker Desktop, to v4.34.2, and when that didn't help, I reverted to v4.34.0.

I've gone so far as to remove Warden entirely from my system and do a fresh install. And I've done "warden env up -d" to fetch new images. No joy.

I would very much appreciate any insights or guidance. 

Thanks,
Kraig

**Comment by navarr** [comment]
This isn't an answer to the problem you're experiencing, but I would highly recommend using [Private Packagist](https://packagist.com) as a mirror for situations when you're pulling in resources from various locations, cost permitting.
